src/compiler.ts

```diff
@@ -8206,6 +8493,20 @@ export class Compiler extends DiagnosticEmitter {
         this.currentType = functionInstance.signature.type;
         return module.i32(index);
       }
+      case ElementKind.CLOSEDLOCAL: {
+        let closedLocal = <ClosedLocal>target;
+        let contextLocal = assert(flow.lookupLocal(CommonNames.this_));
+
+        // TODO: replace this with a class field access, once we are able to construct the class before
+        // compiling
+        return module.load(
+          closedLocal.type.byteSize,
+          true,
```

Comment:
This appears to always perform a signed load potentially leading to wrong values when the type is unsigned, i.e. an `i32.load8_s` expands to a different value than `i32.load8_u` if msb is set.